### PR TITLE
:bug: :recycle: trying to fix i18n with ssr

### PR DESCRIPTION
**Pull request by FabianWilms**
**Description**

All efforts trying to get vue i18n to work with vitepress and SSR/vite failed. It either worked locally or during the build, but not both ways.

So in the end I opted for this way: Building my own translator-Composable which should behave in the same way that vue i18n did.

@MrSebastian feel free to give your feedback, too cause I build this upon your MR

**Reference**

Issues #20

### CODE REVIEW

File: .vitepress/config.mts [hunk 1]
```diff
@@ -104,4 +104,4 @@ export default defineConfig({
       md.use(MarkdownItFootnote);
     }
   }
-})
```
Comment (MrSebastian) [inline .vitepress/config.mts:107]: why that change?

File: .vitepress/components/Translator.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import {useData} from "vitepress";
+import {messages} from "../theme/enhancements/i18n/i18n";
```
Comment (MrSebastian) [inline .vitepress/components/Translator.ts:2]: the file is missing? But i dont get it why it works ...

File: .vitepress/theme/LhmThemeExtension.vue [hunk 3]
```diff
@@ -2,6 +2,7 @@
 <script setup>
 import DefaultTheme from 'vitepress/theme'
 import SoftwareInfoBox from "./SoftwareInfoBox.vue";
+
```
Comment (MrSebastian) [inline .vitepress/theme/LhmThemeExtension.vue:5]: why that change?

File: .vitepress/components/Translator.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+        if(Object.keys(messages).indexOf(lang.value) != -1) {
+            langMap = messages[lang.value];
+        } else {
+            langMap = messages[FALLBACK_LANG];
+        }
+
+        let result = langMap;
+
+        translationKey.split(".").forEach(key => {
```
Comment (MrSebastian) [inline .vitepress/components/Translator.ts:21]: can u make a constant for `.` to make the code more readable?

File: .vitepress/theme/enhancements/i18n/lang/en.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { localizedMessagesType } from "../utility/types";
```
Comment (MrSebastian) [inline .vitepress/theme/enhancements/i18n/lang/en.ts:1]: is missing

File: .vitepress/theme/enhancements/i18n/lang/de.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { localizedMessagesType } from "../utility/types";
```
Comment (MrSebastian) [inline .vitepress/theme/enhancements/i18n/lang/de.ts:1]: is missing

File: .vitepress/components/TagChip.vue [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<script setup>
+
+import {TAG_PROP} from "../theme/enhancements/i18n/utility/tags.ts";
```
Comment (MrSebastian) [inline .vitepress/components/TagChip.vue:3]: missing?

**Review by MrSebastian — CHANGES_REQUESTED** [review]
Maybe u can implement the translator as plugin to prevent that it has to be imported every time.

File: .vitepress/components/TagChip.vue (hunk 7 above)
Comment (FabianWilms) [reply .vitepress/components/TagChip.vue:3]: I think this happened because of the revert-commits. Will fix it!

Comment (FabianWilms) [reply .vitepress/components/TagChip.vue:3]: It's back!

File: .vitepress/components/Translator.ts (hunk 2 above)
Comment (FabianWilms) [reply .vitepress/components/Translator.ts:2]: It's back!

I think it was a problem with merging the commit that reverted the i18n changes. Kind of weird.

File: .vitepress/theme/enhancements/i18n/lang/de.ts (hunk 6 above)
Comment (FabianWilms) [reply .vitepress/theme/enhancements/i18n/lang/de.ts:1]: It's back!

File: .vitepress/theme/enhancements/i18n/lang/en.ts (hunk 5 above)
Comment (FabianWilms) [reply .vitepress/theme/enhancements/i18n/lang/en.ts:1]: It's back!

File: .vitepress/components/Translator.ts (hunk 4 above)
Comment (FabianWilms) [reply .vitepress/components/Translator.ts]: I have 👏

File: .vitepress/theme/LhmThemeExtension.vue (hunk 3 above)
Comment (FabianWilms) [reply .vitepress/theme/LhmThemeExtension.vue]: Is reverted

File: .vitepress/config.mts (hunk 1 above)
Comment (FabianWilms) [reply .vitepress/config.mts]: Is reverted

**Comment by FabianWilms** [comment]
> Maybe u can implement the translator as plugin to prevent that it has to be imported every time.

I will solve this in a new issue, as there are other branches waiting for this i18n feature.

File: .vitepress/components/Translator.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import {useData} from "vitepress";
+import {messages} from "../theme/enhancements/i18n/i18n";
+
+export function useTranslator() {
+
+    const FALLBACK_LANG = 'en';
+    const TRANSLATION_KEY_JSON_PATH_SEPERATOR = '.';
```
Comment (MrSebastian) [inline .vitepress/components/Translator.ts:7]:
```suggestion
    const TRANSLATION_KEY_JSON_PATH_SEPARATOR = '.';
```

File: .vitepress/components/Translator.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+        if(Object.keys(messages).indexOf(lang.value) != -1) {
+            langMap = messages[lang.value];
+        } else {
+            langMap = messages[FALLBACK_LANG];
+        }
+
+        let result = langMap;
+
+        translationKey.split(TRANSLATION_KEY_JSON_PATH_SEPERATOR).forEach(key => {
```
Comment (MrSebastian) [inline .vitepress/components/Translator.ts:22]:
```suggestion
        translationKey.split(TRANSLATION_KEY_JSON_PATH_SEPARATOR ).forEach(key => {
```

**Review by MrSebastian — CHANGES_REQUESTED** [review]
👍 Just a little typo

**Review by MrSebastian — APPROVED** [review]

**Review by klml — APPROVED** [review]

**Pull request merged**